var/spack/repos/builtin/packages/sqlite/package.py

```diff
@@ -77,11 +67,19 @@ class Sqlite(AutotoolsPackage, NMakePackage):
             description="Support loadable extensions",
             when=f"platform={plat}",
         )
-        variant(
-            "rtree", default=True, description="Build with Rtree module", when=f"platform={plat}"
-        )
+
         depends_on("readline", when=f"platform={plat}")
 
+    function_condition = "platform=windows" if is_windows else "+dynamic_extensions"
```

Comment:
Can you put this next to the variant it conditions? I think a comment like

> On windows this variant is always available. On Unix it is only available for `+dynamic_extensions`

since IMO this is trending toward complex for package logic.

Reply:
This has been addressed, let me know if the comment is specific enough